tests/snmp/test_snmp_phy_entity.py

```diff
@@ -0,0 +1,526 @@
+SENSOR_TYPE_PORT_TX_BIAS = 4 * SENSOR_TYPE_MULTIPLE
+
+CHASSIS_SUB_ID = 1
+CHASSIS_MGMT_SUB_ID = MODULE_TYPE_MGMT
+
+# field_name : (name, position)
+PSU_SENSOR_INFO = {
+    'temp': ('Temperature', 1, SENSOR_TYPE_TEMP),
+    'power': ('Power', 2, SENSOR_TYPE_POWER),
+    'current': ('Current', 3, SENSOR_TYPE_CURRENT),
+    'voltage': ('Voltage', 4, SENSOR_TYPE_VOLTAGE),
+}
+
+XCVR_SENSOR_OID_LIST = [SENSOR_TYPE_TEMP,
+                        SENSOR_TYPE_VOLTAGE,
+                        SENSOR_TYPE_PORT_RX_POWER + 1,
+                        SENSOR_TYPE_PORT_RX_POWER + 2,
+                        SENSOR_TYPE_PORT_RX_POWER + 3,
+                        SENSOR_TYPE_PORT_RX_POWER + 4,
+                        SENSOR_TYPE_PORT_TX_BIAS + 1,
+                        SENSOR_TYPE_PORT_TX_BIAS + 2,
+                        SENSOR_TYPE_PORT_TX_BIAS + 3,
+                        SENSOR_TYPE_PORT_TX_BIAS + 4,
+                        SENSOR_TYPE_PORT_TX_POWER + 1,
+                        SENSOR_TYPE_PORT_TX_POWER + 2,
+                        SENSOR_TYPE_PORT_TX_POWER + 3,
+                        SENSOR_TYPE_PORT_TX_POWER + 4]
+
+# Redis Constants
+CHASSIS_KEY = 'chassis 1'
+FAN_DRAWER_KEY_TEMPLATE = 'FAN_DRAWER_INFO|{}'
+FAN_KEY_TEMPLATE = 'FAN_INFO|{}'
+PSU_KEY_TEMPLATE = 'PSU_INFO|{}'
+THERMAL_KEY_TEMPLATE = 'TEMPERATURE_INFO|{}'
+PHYSICAL_ENTITY_KEY_TEMPLATE = 'PHYSICAL_ENTITY_INFO|{}'
+XCVR_KEY_TEMPLATE = 'TRANSCEIVER_INFO|{}'
+XCVR_DOM_KEY_TEMPLATE = 'TRANSCEIVER_DOM_SENSOR|{}'
+
+
+@pytest.fixture(scope="module")
+def snmp_physical_entity_info(duthost, localhost, creds):
+    """
+    Module level fixture for getting physical entity information from snmp fact
+    :param duthost: DUT host object
+    :param localhost: localhost object
+    :param creds: Credential for snmp
+    :return:
+    """
+    return get_entity_mib(duthost, localhost, creds)
+
+
+def get_entity_mib(duthost, localhost, creds):
+    """
+    Get physical entity information from snmp fact
+    :param duthost: DUT host object
+    :param localhost: localhost object
+    :param creds: Credential for snmp
+    :return:
+    """
+    hostip = duthost.host.options['inventory_manager'].get_host(duthost.hostname).vars['ansible_host']
+    snmp_facts = localhost.snmp_facts(host=hostip, version="v2c", community=creds["snmp_rocommunity"])['ansible_facts']
+    entity_mib = {}
+    for oid, info in snmp_facts['snmp_physical_entities'].items():
+        entity_mib[int(oid)] = info
+    return entity_mib
+
+
+def test_fan_drawer_info(duthost, snmp_physical_entity_info):
+    """
+    Verify fan drawer information in physical entity mib with redis database
+    :param duthost: DUT host object
+    :param snmp_physical_entity_info: Physical entity information from snmp fact
+    :return:
+    """
+    keys = redis_get_keys(duthost, STATE_DB, FAN_DRAWER_KEY_TEMPLATE.format('*'))
+    if not keys:
+        pytest.skip('Fan drawer information not exists in DB, skipping this test')
```

Comment:
why skip instead of fail? same comments to below tests.

Reply:
There are platforms such as 2100/2010 who has no physical fan drawer.